build_system/src/prepare.rs

```diff
@@ -0,0 +1,113 @@
+use crate::rustc_info::get_rustc_path;
+use crate::utils::{cargo_install, git_clone, run_command, walk_dir};
+
+use std::fs;
+use std::path::Path;
+
+fn prepare_libcore() -> Result<(), String> {
+    let rustc_path = match get_rustc_path() {
+        Some(path) => path,
+        None => return Err("`rustc` path not found".to_owned()),
+    };
+
+    let parent = match rustc_path.parent() {
+        Some(path) => path,
+        None => return Err(format!("No parent for `{}`", rustc_path.display())),
+    };
+
+    let src_dir = parent.join("../lib/rustlib/src/rust");
+    if !src_dir.is_dir() {
+        return Err("Please install `rust-src` component".to_owned());
+    }
+
+    let dst_dir = Path::new("sysroot_src");
+    if dst_dir.is_dir() {
+        if let Err(e) = fs::remove_dir_all(dst_dir) {
+            return Err(format!("Failed to remove `{}`: {:?}", dst_dir.display(), e));
+        }
+    }
+
+    let dst_dir = dst_dir.join("library");
+    fs::create_dir_all(&dst_dir)
+        .map_err(|e| format!("Failed to folder `{}`: {e:?}", dst_dir.display()))?;
+
+    run_command(&["cp".as_ref(), "-r".as_ref(), src_dir.as_os_str(), dst_dir.as_os_str()], None)?;
+
+    println!("[GIT] init (cwd): `{}`", dst_dir.display());
+    run_command(&["git", "init"], Some(&dst_dir))?;
+    println!("[GIT] add (cwd): `{}`", dst_dir.display());
+    run_command(&["git", "add", "."], Some(&dst_dir))?;
+    println!("[GIT] commit (cwd): `{}`", dst_dir.display());
+
+    // This is needed on systems where nothing is configured.
+    // git really needs something here, or it will fail.
+    // Even using --author is not enough.
+    run_command(&["git", "config", "user.email", "none@example.com"], Some(&dst_dir))?;
+    run_command(&["git", "config", "user.name", "None"], Some(&dst_dir))?;
```

Comment:
`core.autocrlf=false` and `commit.gpgSign=false` may also be necessary depending on existing configuration the user has. This is what I use to apply these to all git commands: https://github.com/bjorn3/rustc_codegen_cranelift/blob/9aa5fbf79c39867f144e5a38c53a8edd2e2069ba/build_system/utils.rs#L170-L187

Reply:
Thanks! Adding that.